src/cli/migrate/nodeVersion/index.ts

```diff
@@ -6,44 +6,126 @@ import fs from 'fs-extra';
 import { log } from '../../../utils/logging';
 import { createDestinationFileReader } from '../../configure/analysis/project';
 
+import { getNode22TypesVersion } from './getNode22TypesVersion';
+import { checkServerlessVersion, checkSkubaType } from './packageJsonChecks';
+
+const DEFAULT_NODE_TYPES = '22.9.0';
+
 type SubPatch = (
   | { files: string; file?: never }
   | { file: string; files?: never }
 ) & {
   test?: RegExp;
   replace: string;
+  id: string;
 };
 
+type VersionResult = {
+  version: string;
+  err: string | undefined;
+};
+
+export const getNode22TypeVersion = (
+  major: number,
+  defaultVersion: string,
+): VersionResult => {
+  try {
+    const version = getNode22TypesVersion(major);
+    if (!version || !/^22.\d+\.\d+$/.test(version)) {
+      throw new Error('No version found');
+    }
+    return {
+      version,
+      err: undefined,
+    };
+  } catch {
+    return {
+      version: defaultVersion,
+      err: 'Failed to fetch latest version, using fallback version',
+    };
+  }
+};
+
+const SHA_REGEX = /(?<=node.*)(@sha256:[a-f0-9]{64})/gm;
+
 const subPatches: SubPatch[] = [
-  { file: '.nvmrc', replace: '<%- version %>\n' },
+  { id: 'nvmrc', file: '.nvmrc', replace: '<%- version %>\n' },
   {
+    id: 'Dockerfile-1',
     files: 'Dockerfile*',
-    test: /^FROM(.*) node:[0-9.]+(\.[^- \n]+)?(-[^ \n]+)?( .+|)$/gm,
-    replace: 'FROM$1 node:<%- version %>$3$4',
+    test: /^FROM(.*) (public.ecr.aws\/docker\/library\/)?node:[0-9.]+(@sha256:[a-f0-9]{64})?(\.[^- \n]+)?(-[^ \n]+)?( .+|)$/gm,
+    replace: 'FROM$1 $2node:<%- version %>$3$5$6',
   },
   {
+    id: 'Dockerfile-2',
     files: 'Dockerfile*',
     test: /^FROM(.*) gcr.io\/distroless\/nodejs\d+-debian(.+)$/gm,
     replace: 'FROM$1 gcr.io/distroless/nodejs<%- version %>-debian$2',
   },
   {
+    id: 'serverless',
     files: 'serverless*.y*ml',
     test: /nodejs\d+.x/gm,
     replace: 'nodejs<%- version %>.x',
   },
   {
+    id: 'cdk',
     files: 'infra/**/*.ts',
     test: /NODEJS_\d+_X/g,
     replace: 'NODEJS_<%- version %>_X',
   },
   {
-    files: '.buildkite/*',
-    test: /image: node:[0-9.]+(\.[^- \n]+)?(-[^ \n]+)?$/gm,
-    replace: 'image: node:<%- version %>$2',
+    id: 'buildkite',
+    files: '**/.buildkite/*',
+    test: /(image: )(public.ecr.aws\/docker\/library\/)?(node:)[0-9.]+(\.[^- \n]+)?(-[^ \n]+)?$/gm,
+    replace: '$1$2$3<%- version %>$5',
+  },
+  {
+    id: 'node-version',
+    files: '.node-version*',
+    test: /(v)?\d+\.\d+\.\d+(.+)?/gm,
+    replace: '$1<%- version %>$2',
+  },
+  {
+    id: 'package-json-1',
+    files: '**/package.json',
+    test: /("@types\/node": ")(\^)?[0-9.]+"/gm,
+    replace: '$1$2<%- version %>"',
+  },
+  {
+    id: 'package-json-2',
+    files: '**/package.json',
+    test: /("engines":\s*{[^}]*"node":\s*">=)(\d+)("[^}]*})(?![^}]*"skuba":\s*{[^}]*"type":\s*"package")/gm,
```

Comment:
This regex is doing what `checkSkubaType()` does, thoughts on keeping both or just using the function?